x-pack/plugin/deprecation/qa/rest/src/main/java/org/elasticsearch/xpack/deprecation/TestDeprecationHeaderRestAction.java

```diff
@@ -97,10 +97,23 @@ public List<Route> routes() {
         return List.of(
             // note: RestApiVersion.current() is acceptable here because this is test code -- ordinary callers of `.deprecated(...)`
             // should use an actual version
-            Route.builder(GET, "/_test_cluster/deprecated_settings").deprecated(DEPRECATED_ENDPOINT, RestApiVersion.current()).build(),
+            Route.builder(GET, "/_test_cluster/deprecated_settings")
+                .deprecatedForRemoval(DEPRECATED_ENDPOINT, RestApiVersion.current())
+                .build(),
+            // TODO: s/deprecated/deprecatedForRemoval when removing `deprecated` method
```

Comment:
Why can't we make this change now?

Reply:
We could ... but I think this is the only remaining test that uses the deprecrated `deprecated` method.